debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Replicator.java

```diff
@@ -146,7 +146,7 @@ public void run() {
                 if (establishConnectionToPrimary()) {
                     if (isInitialSyncExpected()) {
                         recordCurrentOplogPosition();
-                        if (!performInitialSync()) {
+                        if (context.isSnapshotAllowed() && !performInitialSync()) {
```

Comment:
How about adding this check to `isInitialSyncExpected()` instead (i.e. return `false` from there if the snapshot mode is set to `NEVER`)? That way, all that logic for finding out whether a snapshot should be done or not will be in a single place.

Reply:
Yeahh that was my initial idea but then I believe we never have a chance to record the current oplog position and wouldn't that cause issues (it would check from the initial)? I tried running my current test with this change to double-check and [it fails](https://travis-ci.org/debezium/debezium/jobs/494105328). However, since I'm relatively new to this I may be wrong and will definately look into it this weekend.

Reply:
I see. So the issue is that the oplog reading will begin from the earliest position in that case, so picking up the previously inserted record.